cluster/src/dunit/scala/org/apache/spark/transaction/SnapshotGIIDUnitTest.scala

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) 2016 SnappyData, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you
+ * may not use this file except in compliance with the License. You
+ * may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing
+ * permissions and limitations under the License. See accompanying
+ * LICENSE file.
+ */
+package org.apache.spark.transaction
+
+import java.util.Properties
+
+import io.snappydata.cluster.ClusterManagerTestBase
+import io.snappydata.test.dunit.SerializableRunnable
+
+import org.apache.spark.sql.{SaveMode, SnappyContext}
+
+case class Data(col1: Int, col2: Int, col3: Int)
+
+class SnapshotGIIDUnitTest(s: String) extends ClusterManagerTestBase(s) {
+
+  def testColumnTableGII(): Unit = {
+
+    val snc = SnappyContext(sc)
+    val tableName = "TestTable"
+    createTable(snc, tableName, Map("BUCKETS" -> "1", "REDUNDANCY" -> "2"))
+
+    var props = bootProps.clone().asInstanceOf[java.util.Properties]
+    val port = ClusterManagerTestBase.locPort
+
+    def restartServer(props: Properties): SerializableRunnable = new SerializableRunnable() {
+      override def run(): Unit = ClusterManagerTestBase.startSnappyServer(port, props)
+    }
+
+    vm1.invoke(classOf[ClusterManagerTestBase], "stopAny")
+
+    val data = for (i <- 1 to 50) yield (Seq(i, (i + 1), (i + 2)))
+    val rdd = snc.sparkContext.parallelize(data, 2).map(s =>
+      Data(s(0), s(1), s(2)))
+    val dataDF = snc.createDataFrame(rdd)
+    dataDF.write.insertInto(tableName)
+    vm1.invoke(restartServer(props))
+    dataDF.write.insertInto(tableName)
+    val numRows = snc.sql(s"select * from $tableName").collect().length
```

Comment:
To actually assert that GII took place in vm1, we should bring vm2 and vm3 down. Otherwise looks ok.

Reply:
This test is only to ensure all the locks taken during GII are behaving properly. We faced issued from snappy layer when locks were not getting released. Hence added a simple test. 
A more detailed test of SnapshotGII is on SnapshotTxGIIDUnit.